xapian-core/backends/dbfactory_remote.cc

```diff
@@ -57,16 +63,39 @@ Remote::open(const string &program, const string &args,
 	     unsigned timeout_)
 {
     LOGCALL_STATIC(API, Database, "Remote::open", program | args | timeout_);
-    RETURN(Database(new ProgClient(program, args, timeout_ * 1e-3, false, 0)));
+	auto rc = new ProgClient(program, args, timeout_ * 1e-3, false, 0);
+	uuid_to_pid[rc->get_uuid()] = rc->get_child();
+	RETURN(Database(rc));
```

Comment:
More incorrect reindentation.  And yet more below.  Please check your editor settings.